pkg/server/server.go

```diff
@@ -902,7 +902,18 @@ func (s *Server) IsReady(ctx context.Context) (bool, error) {
 	// for now we only depend on the datastore being ready, but in the future
 	// server readiness may also depend on other criteria in addition to the
 	// datastore being ready.
-	return s.datastore.IsReady(ctx)
+
+	status, err := s.datastore.IsReady(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	if status.IsReady {
+		return true, nil
+	}
+
+	s.logger.WarnWithContext(ctx, "datastore health check failed", zap.Any("status", status.Message))
```

Comment:
This log is incorrect, if the health check failed, execution should have entered line 908

Reply:
Updated